src/context/analysisResultStateSlice.ts

```diff
@@ -0,0 +1,311 @@
+  // for local name too.
+  const adminLevelLocalName =
+    adminLayer.adminLevelLocalNames[adminLevel - 1] ||
+    adminLayer.adminLevelLocalNames[0];
+
+  return aggregateData.map((row: any) => {
+    // find feature from admin boundaries that closely matches this api row.
+    // once we find it we can get the corresponding local name.
+    const featureBoundary = find(
+      adminLayerData.features,
+      (feature: any) =>
+        feature.properties[adminLevelName] === row[adminLevelName],
+    );
+
+    const name: string =
+      featureBoundary?.properties?.[adminLevelName] || 'No Name';
+    const localName: string =
+      featureBoundary?.properties?.[adminLevelLocalName] || 'No Name';
+
+    const baselineValue =
+      layerData.find(({ adminKey }) => {
+        return (featureBoundary?.properties?.[
+          adminLayer.adminCode
+        ] as string).startsWith(adminKey);
+      })?.value || 'No Data';
+
+    const tableRow: TableRow = {
+      name,
+      localName,
+      [statistic]: get(row, statistic, 0),
+      baselineValue,
+      coordinates: (featureBoundary?.geometry as any).coordinates[0][0][0], // TODO likely will not keep
+    };
+    return tableRow;
+  });
+}
+
+export type AnalysisDispatchParams = {
+  baselineLayer: NSOLayerProps;
+  hazardLayer: WMSLayerProps;
+  extent: Extent;
+  threshold: ThresholdDefinition;
+  date: ReturnType<Date['getTime']>; // just a hint to developers that we give a date number here, not just any number
+  statistic: AggregationOperations; // we might have to deviate from this if analysis accepts more than what this enum provides
+};
+const apiUrl = 'https://prism-api.ovio.org/stats'; // TODO both needs to be stored somewhere
+// TODO - we need to use the public version generated by the app itself
+const adminJson =
+  'https://prism-admin-boundaries.s3.us-east-2.amazonaws.com/mng_admin_boundaries.json';
+
+export const requestAndStoreAnalysis = createAsyncThunk<
+  AnalysisResult,
+  AnalysisDispatchParams,
+  CreateAsyncThunkTypes
+>('analysisResultState/requestAndStoreAnalysis', async (params, api) => {
+  const {
+    hazardLayer,
+    date,
+    baselineLayer,
+    extent,
+    statistic,
+    threshold,
+  } = params;
+  const baselineData = layerDataSelector(baselineLayer.id)(
+    api.getState(),
+  ) as LayerData<NSOLayerProps>;
+  const adminBoundaries = getBoundaryLayerSingleton();
+  const adminBoundariesData = layerDataSelector(adminBoundaries.id)(
+    api.getState(),
+  ) as LayerData<BoundaryLayerProps>;
+
+  if (!adminBoundariesData) {
+    throw new Error('Boundary Layer not loaded!');
+  }
+  // we force group by to be defined with &
+  // eslint-disable-next-line camelcase
+  const apiRequest: ApiData & { group_by: string } = {
+    geotiff_url: getWCSLayerUrl({
+      layer: hazardLayer,
+      date,
+      extent,
+    }),
+    zones_url:
+      process.env.NODE_ENV === 'production'
```

Comment:
Will do.

A quick summary: 
We are essentially flipping between using the constant amdin boundaries url and the file stored with the frontend. Since the backend cannot access files from localhost during development, we use the constant. 

Otherwise, we construct a URL for the backend to use which is directly from the frontend server.